DEVELOPERS.md

```diff
@@ -2,73 +2,118 @@
+If you want to stop using `yarn` offline, `yarn` will need to be reconfigured to remove
+the offline mirror added by `setup-env.sh`:
+
+    yarn config delete yarn-offline-mirror
 
-    source /usr/share/ovirt-engine-nodejs-modules/setup-env.sh   # to set PATH and ./node_modules directory based on yarn offline cache
-    ./autogen.sh
 
-    export PATH=/usr/share/ovirt-engine-yarn/bin:/usr/share/ovirt-engine-nodejs/bin:$PATH    # please consider adding to ~/.bashrc
+**NOTE:** During development it is not necessary to limit your environment to the offline
+yarn repo provided by `ovirt-engine-nodejs-modules`.  However, it is recommended to run a
+full build to verify no dependency errors will be introduced by your change.
 
-    make    # to create the 'build' directory
-    # or
-    make rpm    # to create (s)rpms under 'tmp.repos'
 
-### Build and install to a local ovirt-engine
+### Development mode
+A primary goal of VM Portal is a quick development cycle (change-build-deploy-check). The
+project uses [webpack-dev-server](http://webpack.github.io/docs/webpack-dev-server.html)
+to accomplish this. To start the server:
+
+    ENGINE_URL=https://my.ovirt.instance/ovirt-engine/ yarn start
+
+When asked, provide a valid user name (in the form of `user@domain`) and password so
+the application can start in the context of a logged in user.
+
+
+### Build
+You can build the static assets from source by:
+
+    # skip if you don't want to use offline mode
+    source /usr/share/ovirt-engine-nodejs-modules/setup-env.sh
 
+    ./autogen.sh
+    make
+
+
+### Build and install to a local development ovirt-engine
 This allows you to run VM Portal deployed directly in an ovirt-engine development installation.
 
+    # skip if you don't want to use offline mode
+    source /usr/share/ovirt-engine-nodejs-modules/setup-env.sh
+
     ./autogen.sh --prefix=/usr --datarootdir=/share
     make all install-data-local DESTDIR=<path_to_engine_development_prefix>
 
-### Development mode
 
-A primary goal of VM Portal is a quick development cycle (change-build-deploy-check). The project uses [webpack-dev-server](http://webpack.github.io/docs/webpack-dev-server.html) to accomplish this. To start the server:
+### Build RPM
+There are at least 3 ways to build the RPM for the project:
+  1. Manually with `make rpm`
 
-    ENGINE_URL=https://my.ovirt.instance/ovirt-engine/ yarn start
+    source /usr/share/ovirt-engine-nodejs-modules/setup-env.sh
+    ./autogen.sh --prefix=/usr --datarootdir=/share
+    make rpm
 
-When asked, provide a valid username (in the form of `user@domain`) and password so
-the application can start in the context of a logged in user.
+  2. Use [mock_runner](https://ovirt-infra-docs.readthedocs.io/en/latest/CI/Using_mock_runner/index.html)
+     to run CI build artifacts locally (this method is cleanest since it runs in a chroot)
+  3. Post the comment "`ci please build`" to the GitHub Pull Request for an on-demand
```

Comment:
worth noting the rpms are auto-generated on push after #950 is merged ?

Reply:
>   4. With [PR#950](https://github.com/oVirt/ovirt-web-ui/pull/950) merged, each pull request push will automatically have RPMs built (see `automation/build.sh`)

@gregsheremeta  - Sound ok?

Reply:
I would leave out "With PR#950 merged" since it already is

Reply:
Reworded just to reference oVirt infra STDCI and swapped order of 3 & 4